src/libraries/Microsoft.PowerFx.Core/Public/DisplayNames/DisplayNameProvider.cs

```diff
@@ -24,12 +24,8 @@ public abstract class DisplayNameProvider
         /// If this isn't supported by a given display name provider, this should return the same as 
         /// <see cref="TryGetLogicalName(DName, out DName)"/>, with the newDisplayName output populated by the first arg. 
         /// </summary>
-        internal abstract bool TryRemapLogicalAndDisplayNames(DName displayName, out DName logicalName, out DName newDisplayName);
+        public abstract bool TryRemapLogicalAndDisplayNames(DName displayName, out DName logicalName, out DName newDisplayName);
 
-        internal abstract ImmutableDictionary<DName, DName> LogicalToDisplayPairs { get; }
-
-        private protected DisplayNameProvider()
-        {
-        }
+        public abstract ImmutableDictionary<DName, DName> LogicalToDisplayPairs { get; }
```

Comment:
>ImmutableDictionary [](http://example.com/codeflow?start=24&length=19)

For public surface, can this just be an IEnumerable<Logical,DisplayName>?

Exposing an Immutable in the contract is too restrictive.

Reply:
#Resolved